client/components/button/index.jsx

```diff
@@ -3,49 +3,43 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import assign from 'lodash/assign';
+import { PropTypes, createElement, PureComponent } from 'react';
 import classNames from 'classnames';
-import noop from 'lodash/noop';
-
-export default React.createClass( {
-
-	displayName: 'Button',
-
-	propTypes: {
-		disabled: React.PropTypes.bool,
-		compact: React.PropTypes.bool,
-		primary: React.PropTypes.bool,
-		scary: React.PropTypes.bool,
-		type: React.PropTypes.string,
-		href: React.PropTypes.string,
-		onClick: React.PropTypes.func,
-		borderless: React.PropTypes.bool
-	},
-
-	getDefaultProps() {
-		return {
-			disabled: false,
-			type: 'button',
-			onClick: noop,
-			borderless: false
-		};
-	},
+import { omit } from 'lodash';
+
+export default class Button extends PureComponent {
+	static propTypes = {
```

Comment:
>do we have some preference to use `static` into the class body vs use Component.propTypes = {}?

I tend to prefer them, since they're more obviously associated with the definition of the class and I think are easier to find since they're typically toward the top of the file. To a lesser extend, they can reduce diffs a bit with `React.createClass` conversions, since line numbers don't change much.

On the flip side, it's a [stage 2 feature](https://github.com/tc39/proposals), so it comes with a certain level of instability and a chance that the feature could be scrapped altogether, which would leave us in a pretty tedious state of having to migrate away from them.